### PR TITLE
Update default engine from nmslib to faiss

### LINKED ISSUE #8589: [DOC] Update default value for engine in k-NN index
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request.
In 2.18, we will be updating default engine from nmslib to fiass. We would like to update the same in documentation
https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#method-definitions

***Version:** List the OpenSearch version to which this issue applies, e.g. 2.14, 2.12--2.14, or all.
2.18
 
**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
https://github.com/opensearch-project/k-NN/issues/2163
PR is raised to make that change in code here https://github.com/opensearch-project/k-NN/pull/2221

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
Starting 2.18, default knn engine is faiss. Updating the documentation accordingly

### Issues Resolved
#8589 

### Version
2.18 onwards

### Frontend features
N/A

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by VijayanB** [comment]
@navneet1v @jmazanec15 are there any other place that needs an update in documentation?

File: _search-plugins/knn/knn-index.md [hunk 1]
```diff
@@ -89,11 +89,11 @@ A method definition will always contain the name of the method, the space_type t
...
 Mapping parameter | Required | Default | Updatable | Description
-:--- | :--- | :--- | :--- | :---
-`name` | true | n/a | false | The identifier for the nearest neighbor method.
-`space_type` | false | l2 | false | The vector space used to calculate the distance between vectors. Note: This value can also be specified at the top level of the mapping.
-`engine` | false | nmslib | false | The approximate k-NN library to use for indexing and search. The available libraries are faiss, nmslib, and Lucene.
-`parameters` | false | null | false | The parameters used for the nearest neighbor method.
+:--- | :--- |:--------| :--- | :---
+`name` | true | n/a     | false | The identifier for the nearest neighbor method.
+`space_type` | false | l2      | false | The vector space used to calculate the distance between vectors. Note: This value can also be specified at the top level of the mapping.
+`engine` | false | faiss   | false | The approximate k-NN library to use for indexing and search. The available libraries are faiss, nmslib, and Lucene.
```
Comment (Naarcha-AWS) [inline _search-plugins/knn/knn-index.md:95]:
```suggestion
`engine` | false | Faiss  | false | The approximate k-NN library to use for indexing and search. The available libraries are Faiss, nmslib, and Lucene.
```

Comment (Naarcha-AWS) [reply _search-plugins/knn/knn-index.md:95]: Faiss wiki uses uppercase.

File: _search-plugins/knn/knn-index.md [hunk 2]
```diff
@@ -89,11 +89,11 @@ A method definition will always contain the name of the method, the space_type t
 (the library) to use, and a map of parameters.
 
 Mapping parameter | Required | Default | Updatable | Description
-:--- | :--- | :--- | :--- | :---
-`name` | true | n/a | false | The identifier for the nearest neighbor method.
-`space_type` | false | l2 | false | The vector space used to calculate the distance between vectors. Note: This value can also be specified at the top level of the mapping.
-`engine` | false | nmslib | false | The approximate k-NN library to use for indexing and search. The available libraries are faiss, nmslib, and Lucene.
-`parameters` | false | null | false | The parameters used for the nearest neighbor method.
+:--- | :--- |:--------| :--- | :---
```
Comment (Naarcha-AWS) [inline _search-plugins/knn/knn-index.md:92]:
```suggestion
:--- | :--- | :--- | :--- | :---
```

File: _search-plugins/knn/knn-index.md [hunk 3]
```diff
@@ -89,11 +89,11 @@ A method definition will always contain the name of the method, the space_type t
...
+`name` | true | n/a     | false | The identifier for the nearest neighbor method.
```
Comment (Naarcha-AWS) [inline _search-plugins/knn/knn-index.md:93]:
```suggestion
`name` | true | n/a | false | The identifier for the nearest neighbor method.
```

File: _search-plugins/knn/knn-index.md [hunk 4]
```diff
@@ -89,11 +89,11 @@ A method definition will always contain the name of the method, the space_type t
...
+`space_type` | false | l2      | false | The vector space used to calculate the distance between vectors. Note: This value can also be specified at the top level of the mapping.
```
Comment (Naarcha-AWS) [inline _search-plugins/knn/knn-index.md:94]:
```suggestion
`space_type` | false | l2 | false | The vector space used to calculate the distance between vectors. Note: This value can also be specified at the top level of the mapping.
```

File: _search-plugins/knn/knn-index.md [hunk 5]
```diff
@@ -89,11 +89,11 @@ A method definition will always contain the name of the method, the space_type t
...
+`engine` | false | faiss   | false | The approximate k-NN library to use for indexing and search. The available libraries are faiss, nmslib, and Lucene.
+`parameters` | false | null    | false | The parameters used for the nearest neighbor method.
```
Comment (Naarcha-AWS) [inline _search-plugins/knn/knn-index.md:96]:
```suggestion
`parameters` | false | null | false | The parameters used for the nearest neighbor method.
```

**Review by Naarcha-AWS — APPROVED** [review]

**Pull request merged**